app/pages/index.vue

```diff
@@ -42,100 +40,95 @@ export default {
       this.subjects = await getCreatedSubjects()
     },
   },
+  middleware: "auth",
 }
 
 // 自分が投稿した質問を配列で返す
-function getVotedSubjects() {
-  return new Promise((resolve) => {
-    firebase.auth().onAuthStateChanged((user) => {
-      db.collection("votes")
-        .where("authId", "==", user.uid)
-        .get()
-        .then((snapshot) => {
-          resolve(convertToSubjectsListFromVotesCollection(snapshot))
-        })
-        .catch((err) => {
-          console.log("Error getting documents", err)
-        })
-    })
+async function getVotedSubjects() {
+  let result = null
+  await firebase.auth().onAuthStateChanged((user) => {
+    result = fetchSubjectsList(user, "votes")
   })
+  return result
 }
 
 // 自分が作成した質問を配列で返す
-function getCreatedSubjects() {
-  return new Promise((resolve) => {
-    firebase.auth().onAuthStateChanged((user) => {
-      db.collection("subjects")
-        .where("authId", "==", user.uid)
-        .get()
-        .then((snapshot) => {
-          resolve(convertToSubjectsListFromSubjectsCollection(snapshot))
-        })
-        .catch((err) => {
-          console.log("Error getting documents", err)
-        })
-    })
+async function getCreatedSubjects() {
+  let result = null
+  await firebase.auth().onAuthStateChanged((user) => {
+    result = fetchSubjectsList(user, "subjects")
   })
+  return result
+}
+
+// firebase から質問一覧を取得して配列に変換して返す
+async function fetchSubjectsList(user, type) {
+  return await db
+    .collection(type)
+    .where("authId", "==", user.uid)
```

Comment:
authId を固定にした場合に動作していることを確認

![image](https://user-images.githubusercontent.com/53972292/95093361-a6692f00-0763-11eb-8e92-e4b97571d3c8.png)

```suggestion
    .where("authId", "==", "1q2a3s45dtuvybiunoip")
```